app/assets/javascripts/admin/modules/document-history-paginator.js

```diff
@@ -0,0 +1,49 @@
+window.GOVUK = window.GOVUK || {}
+window.GOVUK.Modules = window.GOVUK.Modules || {};
+
+(function (Modules) {
+  function DocumentHistoryPaginator (module) {
+    this.module = module
+  }
+
+  DocumentHistoryPaginator.prototype.init = function () {
+    var fetch = window.fetch
+    if (!fetch) { return }
+
+    this.setupEventListeners()
+  }
+
+  DocumentHistoryPaginator.prototype.setupEventListeners = function () {
+    var module = this.module
+    var newerLink = module.querySelector('.app-view-document-history-tab__newer-pagination-link')
+    var olderLink = module.querySelector('.app-view-document-history-tab__older-pagination-link')
+
+    if (newerLink) {
+      this.addLinkEventListener(newerLink)
+    }
+
+    if (olderLink) {
+      this.addLinkEventListener(olderLink)
+    }
+  }
+
+  DocumentHistoryPaginator.prototype.addLinkEventListener = function (link) {
+    var module = this.module
+
+    link.addEventListener('click', function (e) {
+      e.preventDefault()
+
+      window.fetch(new URL(document.location.origin + link.dataset.remotePagination))
+        .then(function (response) { return response.text() })
+        .catch(function () { window.location.replace(new URL(link.href)) })
```

Comment:
I'm not sure if `window.location.replace()` is the method you want to be using here.

The [W3Schools definition](https://www.w3schools.com/jsref/met_loc_replace.asp) explains:

> `replace()` removes the current URL from the document history.
>
> With `replace()` it is not possible to use "back" to navigate back to the original document.

I think the expected behaviour is that you'd still be able to use the "back" button to go back a page – i.e. the same as if JavaScript was disabled. In this case, you probably want to set `window.location.href` instead:

```js
window.location.href = new URL(link.href)
```

And since you're not manipulating the URL itself, just passing it through, you'd be fine to just pass it through as a String. (I actually didn't know about [URL objects](https://javascript.info/url) before seeing this PR! 😅)

```suggestion
        .catch(function () { window.location.href = link.href })
```